crates/core/src/db/datastore/locking_tx_datastore/mod.rs

```diff
@@ -2486,50 +2566,38 @@ mod tests {
-                index_row(4, 4, 0, "constraint_id_idx", true),
-                index_row(5, 1, 0, "idx_ct_columns_table_id", false),
+                index_row(0, 0, &[0], "table_id_idx", true),
+                index_row(1, 3, &[0], "index_id_idx", true),
+                index_row(2, 2, &[0], "sequences_id_idx", true),
+                index_row(3, 0, &[1], "table_name_idx", true),
+                index_row(4, 4, &[0], "constraint_id_idx", true),
+                index_row(5, 1, &[0], "idx_ct_columns_table_id", false),
             ]
         );
-        let sequence_rows = datastore
-            .iter_mut_tx(&ExecutionContext::default(), &tx, ST_SEQUENCES_ID)?
-            .map(|x| StSequenceRow::try_from(x.view()).unwrap().to_owned())
-            .sorted_by_key(|x| x.sequence_id)
-            .collect::<Vec<_>>();
+        let sequence_rows = query.scan_st_sequences()?;
         #[rustfmt::skip]
         assert_eq!(
             sequence_rows,
             vec![
-                StSequenceRow { sequence_id: 0.into(), sequence_name: "table_id_seq".to_string(), table_id: 0.into(), col_id: 0.into(), increment: 1, start: 6, min_value: 1, max_value: 4294967295, allocated: 4096 },
-                StSequenceRow { sequence_id: 1.into(), sequence_name: "sequence_id_seq".to_string(), table_id: 2.into(), col_id: 0.into(), increment: 1, start: 4, min_value: 1, max_value: 4294967295, allocated: 4096 },
-                StSequenceRow { sequence_id: 2.into(), sequence_name: "index_id_seq".to_string(), table_id: 3.into(), col_id: 0.into(), increment: 1, start: 6, min_value: 1, max_value: 4294967295, allocated: 4096 },
-                StSequenceRow { sequence_id: 3.into(), sequence_name: "constraint_id_seq".to_string(), table_id: 4.into(), col_id: 0.into(), increment: 1, start: 1, min_value: 1, max_value: 4294967295, allocated: 4096 },
+                // sequence_id, sequence_name, table_id, col_pos, start
+                sequence_row(0, "table_id_seq", 0,  0, 6),
+                sequence_row(1, "sequence_id_seq", 2, 0, 4),
+                sequence_row(2, "index_id_seq", 3, 0, 6),
+                sequence_row(3, "constraint_id_seq", 4, 0, 1),
             ]
         );
-        let constraints_rows = datastore
-            .iter_mut_tx(&ExecutionContext::default(), &tx, ST_CONSTRAINTS_ID)?
-            .map(|x| StConstraintRow::try_from(x.view()).unwrap().to_owned())
-            .sorted_by_key(|x| x.constraint_id)
-            .collect::<Vec<_>>();
-
+        let constraints_rows = query.scan_st_constraints()?;
         #[rustfmt::skip]
         assert_eq!(
             constraints_rows,
             vec![
-                StConstraintRow{ constraint_id: 5.into(), constraint_name: "ct_columns_table_id".to_string(), kind:  ColumnIndexAttribute::INDEXED, table_id: 1.into(), columns: vec![0.into()] },
+                constraint_row(5, "ct_columns_table_id", 1, ColumnIndexAttribute::INDEXED, &[0]),
```

Comment:
I mildly don't like that this loses some clarity at call sites, 5 and 1 are like, just numbers now.

Reply:
I provided the same feedback when @Centril proposes this in another PR. This explicitness is by design. IMO we should revert this change and the other changes like it.